graphqlapi/src/main/java/io/stargate/graphql/schema/graphqlfirst/fetchers/deployed/UpdateFetcher.java

```diff
@@ -43,43 +44,51 @@ public UpdateFetcher(UpdateModel model, MappingModel mappingModel) {
   }
 
   @Override
-  protected Boolean get(
+  protected Map<String, Object> get(
       DataFetchingEnvironment environment, DataStore dataStore, StargateGraphqlContext context)
       throws UnauthorizedException {
+    DataFetchingFieldSelectionSet selectionSet = environment.getSelectionSet();
 
     EntityModel entityModel = model.getEntity();
     Keyspace keyspace = dataStore.schema().keyspace(entityModel.getKeyspaceName());
-    Map<String, Object> input = environment.getArgument(model.getEntityArgumentName());
+
+    Map<String, Object> response = new LinkedHashMap<>();
+    // We're either getting the values from a single entity argument, or individual PK field
+    // arguments:
+    java.util.function.Predicate<String> hasArgument;
+    Function<String, Object> getArgument;
+    if (model.getEntityArgumentName().isPresent()) {
+      Map<String, Object> entity = environment.getArgument(model.getEntityArgumentName().get());
+      hasArgument = entity::containsKey;
+      getArgument = entity::get;
+    } else {
+      hasArgument = environment::containsArgument;
+      getArgument = environment::getArgument;
+    }
 
     List<BuiltCondition> whereConditions =
         bindWhere(
             entityModel.getPrimaryKeyWhereConditions(),
             entityModel,
-            input::containsKey,
-            input::get,
+            hasArgument,
+            getArgument,
             keyspace);
+    List<BuiltCondition> ifConditions =
+        bindIf(model.getIfConditions(), hasArgument, getArgument, keyspace);
+    boolean isLwt = !ifConditions.isEmpty() || model.ifExists();
 
     Collection<ValueModifier> modifiers = new ArrayList<>();
-    for (FieldModel column : entityModel.getRegularColumns()) {
-      String graphqlName = column.getGraphqlName();
-      if (input.containsKey(graphqlName)) {
-        Object graphqlValue = input.get(graphqlName);
-        modifiers.add(
-            ValueModifier.set(
-                column.getCqlName(), toCqlValue(graphqlValue, column.getCqlType(), keyspace)));
-      }
-    }
-    if (modifiers.isEmpty()) {
-      throw new IllegalArgumentException(
-          "Input object must have at least one non-PK field set for an update");
-    }
+    prepopulateWithInputData(
```

Comment:
I am using a similar pattern to `InsertFetcher` to pre-populate response with data from the input selection. Later, they are overridden if the LWT (IF) fails.